src/core/cluster/quorum.rs

```diff
@@ -250,37 +274,36 @@ impl Cluster for Quorum {
             ); // TODO move meta
             return Ok(answer.into_inner());
         } else if errors.is_empty() {
-            debug!("GET[{}] data not found", key);
-            return Err(Error::key_not_found(key));
+            debug!("GET[{}] data not found on any node in regular dir", key);
+        } else {
+            error!("{}", errors);
         }
-        trace!("appropriate nodes didn't get successful results, lookup in other nodes aliens");
-        debug!("GET[{}] no success result", key);
+        trace!("GET[{}] ~~~LOOKUP REMOTE NODES ALIEN~~~", key);
 
-        let sup_nodes: Vec<_> = self
-            .get_support_nodes(all_nodes.iter().map(Node::index), 0)?
-            .into_iter()
-            .cloned()
-            .collect(); // @TODO take from config
-
-        debug!("GET[{}]: Sup nodes for fan out: {:?}", key, &sup_nodes);
-
-        let second_attempt = Self::get_all(key, &sup_nodes, GetOptions::new_alien()).await;
-        debug!("GET[{}] cluster ans sup: {:?}", key, second_attempt);
+        let local_node = self.mapper.local_node_name();
+        let target_nodes = self
+            .mapper
+            .nodes()
+            .iter()
+            .filter(|node| node.name() != local_node);
+        // @TODO don't wait for all answers, return on first success
+        let results = Self::get_all(key, target_nodes, GetOptions::new_alien()).await;
```

Comment:
we already sent request for ALL before. do we need to query alien directly?

Reply:
fixed